File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int s = nums.size();
        int c = 0;
        sort(nums.begin(), nums.end());
        for(int i=0;i<s-1;i++){
            if(nums[i] != nums[i+1]){
                c = max(i+1-c, c);
                if(c < (s/2 + s%2)) continue;
                return nums[i];
            }
        }
        return nums[s-1];
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int size = nums.length;
        int count = 0;
        int current = 0;
        for(int num : nums){
            if(count == 0){
                current = num;
            }
            if(current == num) count++;
            else count--;
        }
        return current;
    }
}